paddle/fluid/framework/ir/conv_bn_fuse_pass.cc

```diff
@@ -0,0 +1,333 @@
+    y[i] = f(a[i], b[i]);
+  }
+  return vec_y;
+}
+
+template <typename BinaryOperation>
+LoDTensor tensor_apply_eltwise_broadcast(LoDTensor vec_a, LoDTensor vec_b,
+                                         BinaryOperation f) {
+  PADDLE_ENFORCE_GT(vec_a.dims().size(), vec_b.dims().size());
+  PADDLE_ENFORCE_EQ(vec_a.dims()[0], vec_b.dims()[0]);
+  LoDTensor vec_y;
+  vec_y.Resize(vec_a.dims());
+  float* a = vec_a.mutable_data<float>(platform::CPUPlace());
+  float* b = vec_b.mutable_data<float>(platform::CPUPlace());
+  float* y = vec_y.mutable_data<float>(platform::CPUPlace());
+  size_t b_size = vec_b.dims()[0];
+  for (int64_t i = 0; i < vec_a.numel(); ++i) {
+    y[i] = f(a[i], b[i % b_size]);
+  }
+  return vec_y;
+}
+
+std::unique_ptr<ir::Graph> ConvBNFusePass::ApplyImpl(
+    std::unique_ptr<ir::Graph> graph) const {
+  PADDLE_ENFORCE(graph.get());
+  FusePassBase::Init(name_scope_, graph.get());
+
+  auto* scope = param_scope();
+  PADDLE_ENFORCE(scope);
+
+  GraphPatternDetector gpd;
+  auto* conv_input =
+      gpd.mutable_pattern()
+          ->NewNode(patterns::PDNodeName(name_scope_, "conv_input"))
+          ->AsInput()
+          ->assert_is_op_input("conv2d", "Input");
+  patterns::ConvBN conv_bn_pattern(gpd.mutable_pattern(), name_scope_);
+  conv_bn_pattern(conv_input, false /*with_eltwise_add*/);
+
+  int found_conv_bn_count = 0;
+  auto handler = [&](const GraphPatternDetector::subgraph_t& subgraph,
+                     Graph* g) {
+    VLOG(4) << "handle ConvBN fuse";
+    // OPERATORS
+    GET_IR_NODE_FROM_SUBGRAPH(conv, conv, conv_bn_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(batch_norm, batch_norm, conv_bn_pattern);
+    // CONV inputs
+    GET_IR_NODE_FROM_SUBGRAPH(conv_weight, conv_weight, conv_bn_pattern);
+    // CONV outputs
+    GET_IR_NODE_FROM_SUBGRAPH(conv_out, conv_out, conv_bn_pattern);
+    // BN inputs
+    GET_IR_NODE_FROM_SUBGRAPH(bn_scale, bn_scale, conv_bn_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(bn_bias, bn_bias, conv_bn_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(bn_mean, bn_mean, conv_bn_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(bn_variance, bn_variance, conv_bn_pattern);
+    // BN outputs
+    GET_IR_NODE_FROM_SUBGRAPH(bn_out, bn_out, conv_bn_pattern);  // Out
+    GET_IR_NODE_FROM_SUBGRAPH(bn_mean_out, bn_mean_out, conv_bn_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(bn_variance_out, bn_variance_out,
+                              conv_bn_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(bn_saved_mean, bn_saved_mean, conv_bn_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(bn_saved_variance, bn_saved_variance,
+                              conv_bn_pattern);
+
+    // Create y (bias) variable
+    VarDesc y_desc(patterns::PDNodeName(name_scope_, "eltwise_y_in"));
+    auto* y_var_node = g->CreateVarNode(&y_desc);
+    auto* bias_y_tensor =
+        scope->Var(y_var_node->Name())->GetMutable<LoDTensor>();
+    std::fill_n(bias_y_tensor->mutable_data<float>(platform::CPUPlace()),
```

Comment:
We don't need to fill in the data with initialized value, since it will be re-initialized later.

Reply:
Do you mean that I wrote fill_n twice?